packages/grid/x-data-grid/src/joy/joySlots.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from 'react';
+import JoyCheckbox from '@mui/joy/Checkbox';
+import JoyInput from '@mui/joy/Input';
+import JoyFormControl from '@mui/joy/FormControl';
+import JoyFormLabel from '@mui/joy/FormLabel';
+import type { UncapitalizeObjectKeys } from '../internals/utils';
+import type { GridSlotsComponent, GridSlotsComponentsProps } from '../models';
+
+const Checkbox = React.forwardRef<
+  HTMLElement,
+  NonNullable<GridSlotsComponentsProps['baseCheckbox']>
+>(
+  (
+    { touchRippleRef, inputProps, onChange, color, size, checked, sx, value, inputRef, ...props },
```

Comment:
I didn't pass the `sx` prop to `JoyCheckbox` because of the type mismatch.
This is not good since it won't be possible to override styles using `slotProps.baseCheckbox`, but I don't know how to solve the typing issue in this case

Reply:
I would expect that this component only needs these props:

```suggestion
    { onChange, checked, value, inputRef, indeterminate, ...props },
```

It almost feels like we need a version of the slots for Material UI. It would add overhead on the runtime and bundle size, so not sure that it's the best idea but to consider if there are props we can normalize more.

Reply:
The destructured props are here to avoid passing them to the component when spreading `props` to it.